common/scripts/babel-conditional-preprocess.js

```diff
@@ -63,62 +65,68 @@ exports.default = babelHelper.declare((_api, opts) => {
       // As this only applies to TypeScript types, it is safe from a code-flow perspective: This does not enable any new
       // conditional business logic flows.
       TSType(path) {
-        Handle(path, annotations);
+        Handle(path, match, featurePrefixes);
       },
 
       TSDeclareMethod(path) {
-        Handle(path, annotations);
+        Handle(path, match, featurePrefixes);
       },
 
       Expression(path) {
-        Handle(path, annotations);
+        Handle(path, match, featurePrefixes);
       },
 
       ClassMethod(path) {
-        Handle(path, annotations);
+        Handle(path, match, featurePrefixes);
       },
 
       ClassProperty(path) {
-        Handle(path, annotations);
+        Handle(path, match, featurePrefixes);
       },
     }
   };
 });
 
-function Handle(path, annotations) {
+function Handle(path, match, featurePrefixes) {
   let { node } = path;
-  let removed = false;
-
-  for (const annotation of annotations) {
-    const {
-      match
-    } = annotation;
-
-    if (node.leadingComments && node.leadingComments.length > 0) {
-      for (const comment of node.leadingComments) {
-        if (comment.value.includes(match)) {
-          removed = true;
-        }
-        if (removed) {
-          comment.ignore = true;
-          // Comment will be inherit to next line even it is set to 'ignore'
-          // this will cause some unexpected removals of code
-          // clear the comment to ensure nothing gets wrong
-          comment.value = '';
-        }
-      }
-    }
 
-    if (removed) {
-      // We cannot remove Expression in JSXExpressionContainer cause it is not correct for AST
-      // Replacing it with jSXEmptyExpression will get us the same result
-      // There will always be only one expression under JSXExpressionContainer
-      if (path?.container?.type === 'JSXExpressionContainer') {
-        path.replaceWith(t.jSXEmptyExpression());
-      } else {
-        path.remove();
-      }
+  const shouldRemove = node.leadingComments && node.leadingComments.some((comment) => containsDirective(comment, match, featurePrefixes));
```

Comment:
Why is this needed?

Reply:
The `forEach` just modifies each comment that contains a directive.

This check is an early return that skips other logic as well (JSX handling for now).